src/content.ts

```diff
@@ -1120,11 +1120,12 @@ function updatePreviewBar(): void {
 async function whitelistCheck() {
     const whitelistedChannels = Config.config.whitelistedChannels;
 
-    const getChannelID = () => videoInfo?.videoDetails?.channelId
-        ?? document.querySelector(".ytd-channel-name a")?.getAttribute("href")?.replace(/\/.+\//, "") // YouTube
-        ?? document.querySelector(".ytp-title-channel-logo")?.getAttribute("href")?.replace(/https:\/.+\//, "") // YouTube Embed
-        ?? document.querySelector("a > .channel-profile")?.parentElement?.getAttribute("href")?.replace(/\/.+\//, "") // Invidious
-        ?? document.querySelector("a.slim-owner-icon-and-title")?.getAttribute("href")?.replace(/\/.+\//, ""); // Mobile YouTube
+    const getChannelID = () =>
+        (document.querySelector("a.ytd-video-owner-renderer") // YouTube
+        ?? document.querySelector("a.ytp-title-channel-logo") // YouTube Embed
+        ?? document.querySelector(".channel-profile #channel-name")?.parentElement.parentElement // Invidious
```

Comment:
```suggestion
        ?? document.querySelector("a > .channel-profile > #channel-name")?.parentElement.parentElement // Invidious
```

Reply:
The `onInvidious` check looks good, but iirc the `#channel-name` element wasn't always an immediate child of `.channel-profile`, and that wasn't always an immediate child of an `a` tag.

If you can verify that is indeed the case, then it makes sense to include this change. I can't check it until tomorrow night at the earliest.

Reply:
If that's not the case, I don't think we can rely on `.parentElement.parentelement`

Reply:
Ah, you're right. This should be merged then - I forgot about the `parentElement`.

Just wanted to make sure there was justification for the direct-child selector.